lms/djangoapps/instructor/lti_connection.py

```diff
@@ -0,0 +1,140 @@
+"""
+Send grades to an OpenEdX LTI component
+"""
+import base64
+import hashlib
+import json
+import logging
+import urllib
+
+import mock
+from oauthlib.oauth1 import Client  # pylint: disable=F0401
+import requests
+
+from django.utils.translation import ugettext as _
+
+log = logging.getLogger(__name__)
+
+
+def validate_lti_passport(key, secret, url):
+    """
+    Sanity-check LTI passport credentials
+    """
+    resp = _send_lti_2_json_request('GET', url, key, secret)
+    if resp.status_code != 200:
+        raise LTIConnectionError(
+            _("LTI passport sanity check failed. Your lti_key ({key}) or lti_secret ({secret}) are probably incorrect.").format(
+                key=key,
+                secret=secret,
+            )
+        )
+
+
+def post_grade(url_base, key, secret, grade_row):
+    """
+    Post a grade to the LTI endpoint
+    """
+    comment = ''
+    uid, anon_id, email, grade, total = grade_row[:5]
+    if len(grade_row) >= 6:
+        comment = grade_row[5]
+    if anon_id == '':
+        raise LTIConnectionError(
+            _("GRADE POST FAILED: user id {user_id} (email: {email}) is missing a proper anonymized id").format(
+                user_id=uid,
+                email=email,
+            )
+        )
+
+    url = url_base + anon_id
+
+    payload = {
+        '@context': 'http://purl.imsglobal.org/ctx/lis/v2/Result',
+        '@type': 'Result',
+        'comment': comment,
+        'resultScore': grade / total,
```

Comment:
if total == 0, an exception is already raise in line 33, so we are already accounting for this